.test.mk

```diff
@@ -123,6 +123,36 @@ test-release-asan: LUAJIT_TEST_ENV = LSAN_OPTIONS=suppressions=${PWD}/asan/lsan.
                                                   unmap_shadow_on_exit=1
 test-release-asan: build run-luajit-test run-test
 
+# Debug ASAN build
+
+.PHONY: test-debug-asan
+test-debug-asan: CMAKE_PARAMS = -DCMAKE_BUILD_TYPE=Debug \
+                                  -DENABLE_WERROR=ON \
+                                  -DENABLE_ASAN=ON \
+                                  -DENABLE_UB_SANITIZER=ON \
+                                  -DFIBER_STACK_SIZE=1280Kb \
+                                  -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION=OFF \
+                                  -DENABLE_FUZZER=ON \
+                                  -DTEST_BUILD=ON
+test-debug-asan: TEST_RUN_ENV = ASAN=ON \
+                                LSAN_OPTIONS=suppressions=${PWD}/asan/lsan.supp \
+                                ASAN_OPTIONS=heap_profile=0:unmap_shadow_on_exit=1:$\
+                                             detect_invalid_pointer_pairs=1:symbolize=1:$\
+                                             detect_leaks=1:dump_instruction_bytes=1:$\
+                                             print_suppressions=0
+test-debug-asan: LUAJIT_TEST_ENV = LSAN_OPTIONS=suppressions=${PWD}/asan/lsan.supp \
+                                   ASAN_OPTIONS=detect_invalid_pointer_pairs=1:$\
+                                                detect_leaks=1:$\
+                                                dump_instruction_bytes=1:$\
+                                                heap_profile=0:$\
+                                                print_suppressions=0:$\
+                                                symbolize=1:$\
+                                                unmap_shadow_on_exit=1
+test-debug-asan: TEST_RUN_PARAMS += --test-timeout 620 \
```

Comment:
You should leave a comment about increasing timeouts.

Reply:
Done.

```diff
diff --git a/.test.mk b/.test.mk
index ed12442d2..dd81c17ad 100644
--- a/.test.mk
+++ b/.test.mk
@@ -136,6 +136,8 @@ test-debug-asan: CMAKE_PARAMS = ${CMAKE_PARAMS_ASAN} \
                                 -DFIBER_STACK_SIZE=1280Kb
 test-debug-asan: TEST_RUN_ENV = ${TEST_RUN_ENV_ASAN}
 test-debug-asan: LUAJIT_TEST_ENV = ${LUAJIT_TEST_ENV_ASAN}
+# Increase timeouts as some tests in ASAN debug build take quite a lot
+# of time to finish.
 test-debug-asan: TEST_RUN_PARAMS += --test-timeout 620 \
                                     --no-output-timeout 630 \
                                     --server-start-timeout 610
```